tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -525,26 +574,39 @@ void TRTEngineOp::ExecuteNativeSegment(OpKernelContext* ctx,
   opts.rendezvous = ctx->rendezvous();
   opts.cancellation_manager = ctx->cancellation_manager();
   opts.runner = ctx->runner();
-  inputs.reserve(ctx->num_inputs());
+  native_inputs.reserve(ctx->num_inputs());
   for (int i = 0; i < ctx->num_inputs(); i++) {
-    inputs.push_back(ctx->input(i));
+    if (ctx->input_dtype(i) != DT_INT32) {
+      native_inputs.push_back(ctx->input(i));
+    } else {
+      OP_REQUIRES_OK_ASYNC(ctx, CopyToHost(ctx, &native_inputs, i),
+                           dummy_async_helper);
+    }
   }
   VLOG(1) << "Executing native segment: " << name();
   // Increment the reference count of the async_helper by 1. When the native
-  // segment finishes execution asynchronously, we decrement the reference count
-  // of the object.
+  // segment finishes execution asynchronously, we decrement the reference
```

Comment:
I think we are missing synchronization right after the for loop. we need to add something like this:

if (we issue any cudaMemcpyAsync in the for loop)
      cudaStreamSynchronize(*stream);